src/firmware/shared/math/matrix_test.cpp

```diff
@@ -0,0 +1,89 @@
+extern "C"
+{
+#include "firmware/shared/math/matrix.h"
+}
+
+#include <gtest/gtest.h>
+
+TEST(MatrixTest, getAndSetMatrixValues)
+{
+    Matrix_t* matrix = shared_matrix_create(3, 4);
+
+    unsigned int col = 2;
+    unsigned int row = 3;
+    float value      = -2.22f;
+    shared_matrix_setValueAtIndex(row, col, value, matrix);
+
+    EXPECT_EQ(shared_matrix_getValueAtIndex(row, col, matrix), value);
+
+    shared_matrix_setValueAtIndex(row - 1, col + 1, value - 20, matrix);
+
+    EXPECT_EQ(shared_matrix_getValueAtIndex(row - 1, col + 1, matrix), value - 20);
+
+    shared_matrix_destroy(matrix);
+}
+
+TEST(MatrixTest, matrixMultiplication)
+{
+    Matrix_t* matrix = shared_matrix_create(3, 4);
+
+    shared_matrix_setValueAtIndex(1, 1, 1, matrix);
+    shared_matrix_setValueAtIndex(1, 2, 2, matrix);
+    shared_matrix_setValueAtIndex(1, 3, 3, matrix);
+    shared_matrix_setValueAtIndex(1, 4, 4, matrix);
+    shared_matrix_setValueAtIndex(2, 1, 5, matrix);
+    shared_matrix_setValueAtIndex(2, 2, 6, matrix);
+    shared_matrix_setValueAtIndex(2, 3, 7, matrix);
+    shared_matrix_setValueAtIndex(2, 4, 8, matrix);
+    shared_matrix_setValueAtIndex(3, 1, 9, matrix);
+    shared_matrix_setValueAtIndex(3, 2, 10, matrix);
+    shared_matrix_setValueAtIndex(3, 3, 11, matrix);
+    shared_matrix_setValueAtIndex(3, 4, 12, matrix);
```

Comment:
can we make a nice constructor instead of doing this? For example, pass in n_cols, n_rows, and an array of values to initialize with

Reply:
Or an array of arrays so it _looks_ like the matrix :eyes: